### PR TITLE
Add Athena adapter support

### PR DESCRIPTION
**Pull request by nathanjcochran**
Adds Athena adapter support for dbt versions 1.0, 1.1, 1.2, 1.3, and 1.4.

Note that I ran `poetry lock` with the `--no-upgrade` flag, which avoids unnecessarily updating existing packages in the lock file, because I wanted to keep the scope of the changes limited to adding the `dbt-athena-community` adapter, as much as possible. That being said, in some cases, some shared packages had to be updated to satisfy the version constraints for `dbt-athena-community` (for example, the locked version of `dbt-core` was upgraded from `1.4.4` to `1.4.6` for the `1.4` adapters). Let me know if you think I should just go ahead and run `poetry lock` without the `--no-update` flag to update everything to the latest compatible versions, though. It would probably be fine, but it resulted in much larger changes to the lock file when I tried it.

I also made some other small updates to the code in this branch, including adding a new `/bin/build.sh` script that can be used to build images locally, so they can be tested without pushing them to ghcr.

Still in the process of testing each of the athena adapters, but wanted to put this PR out there to get any early feedback.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -15,6 +13,9 @@ RUN apt-get update \
   && groupadd -f -g ${GID} -r dbt && useradd -g dbt -l -m -r -u ${UID} dbt \
   && python3 -m pip install -U wheel
 
+ARG DBT_VERSION
+ARG DBT_ADAPTER
```
Comment (nathanjcochran) [inline Dockerfile:17]: `ARG` instructions invalidate the build cache for subsequent `RUN` instructions. Moving this below the run instruction above (which doesn't use either of these args) allows that layer to be re-used when building images for different adapters/versions.

File: bin/generate_requirements.sh [hunk 2]
```diff
@@ -26,7 +26,7 @@ if [ ! -d "${requirements_dir}" ]; then
   exit 1
 fi
 
-if grep -Fxq "$adapter" "${requirements_dir}/${version}/pyproject.toml"; then
+if grep -Fxq "$adapter" "${requirements_dir}/pyproject.toml"; then
```
Comment (nathanjcochran) [inline bin/generate_requirements.sh:29]: The `${version}` variable is already included in `${requirements_dir}` above. It was being duplicated here.

**Review by MasterOdin — APPROVED** [review]

**Comment by MasterOdin** [comment]
I think it's probably safer to use `--no-upgrade` as unfortunately python libraries often don't put strict upper bounds on dependencies, and so it ends up possible to bring in a new major version of a package that breaks things (which is why we're pinning some transitive dependencies for older dbt versions).

**Review by MasterOdin — APPROVED** [review]
Thanks for updating the README.

**Pull request merged**